### ISSUE TITLE
Add missing Show instances

### ISSUE THREAD

**Issue by ceedubs** [typelevel/cats#607]
Many data types already have `Show` instances (and more will after #600, #601, and #602 are merged). However, there are still a few that are missing. Two that I see at the moment are ~~`Prod`~~ and `WriterT`. We should add those and any others we can find that are missing.

Edit: removed `Prod` from the scope of this issue per https://github.com/non/cats/issues/607#issuecomment-155074440.

**Comment by mikejcurry** [comment]
I was going to add an instance for `Prod`; and learn the functionality at the same time, but it turned out to be a bit like peeling an onion. Less the tears part and more the layers part, but... :-)

Anyway, the thing with `Prod` specifically, and maybe this deserves its own issue, is that it is constructed to be lazy and it feels wrong for the `Show` instance to bypass this and compute the value. So, I figured an `Eval` instance for `Show` which respected the laziness of `Eval` might be useful.

Before I go too deep down this road, the question I have is whether a `Show` for `Eval` makes sense. E.g. for something like `Now` it would show _Now(value)_ and something like Later it could show the value if already computed and could show something like _Later(() => ResultType)_ if the value has not been computed.

The next layer of the onion is that there are `Show` instances for `BigInt` and `BigDecimal` but not for primitives such as `Int`, `Char` etc (as far as I could see) - does it make sense to add these also? 

My feeling is that all of these make sense at some level, and that `Show` is pretty much universally applicable as you may want to `Show` pretty much anything.

Let me know what you think.

**Comment by ceedubs** [comment]
@mikejcurry I'm about to go somewhere, but there's one simple part of this that I can address first :). The `Show` instances for primitives are [here](https://github.com/non/cats/blob/master/core/src/main/scala/cats/std/anyval.scala).

**Comment by mikejcurry** [comment]
Aah ok, I missed those for some reason. Sorry about that.

**Comment by ceedubs** [comment]
@mikejcurry Thank you for brining up the rabbit hole you went down with `Prod`. I agree that it's awkward, and I've created #615 to address that. Let's drop `Prod` from the scope of this issue.

**Comment by matt-martin** [comment]
I've tried to add a Show instance for WriterT with the following pull request: https://github.com/typelevel/cats/pull/928.  Just trying to get my feet wet here with some "low-hanging fruit" so I'd welcome any and all feedback.

**Comment by rabbitonweb** [comment]
Since #928 is merged, shouldn't this issue be closed?

**Comment by ceedubs** [comment]
@rabbitonweb I suspect there are more missing, but this is a pretty broad/vague issue, so if we find any we should probably create a separate issue for them. Thanks for the heads up - I'll close this out.

**Comment by rabbitonweb** [comment]
@ceedubs I'm sort of looking for a low hanging fruit - an entrypoint to contribution to cats, maybe if u could find one for, creating a Show instance would be a good starting point
